### PR TITLE
OCD-4237 - Upgrade Spring Framework

### PR DESCRIPTION
**Pull request by tmy1313**

### CODE REVIEW

File: chpl/chpl-api/src/main/java/gov/healthit/chpl/filter/DeprecatedEndpointUsageFilter.java [hunk 1]
```diff
@@ -74,6 +74,11 @@ public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
             return;
         }
 
+        //if (!ServletRequestPathUtils.hasParsedRequestPath(request)) {
```
Comment (kekey1) [inline chpl/chpl-api/src/main/java/gov/healthit/chpl/filter/DeprecatedEndpointUsageFilter.java]: Can this be removed?

File: chpl/chpl-api/src/main/java/gov/healthit/chpl/filter/DeprecatedEndpointUsageFilter.java [hunk 2]
```diff
@@ -91,7 +96,12 @@ public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
     }
 
     private void logDeprecatedApiUsage(RequestMappingInfo requestMapping, HandlerMethod handlerMethod, HttpServletRequest request) {
-        Set<String> matchingUrlPatterns = requestMapping.getPatternsCondition().getPatterns();
+        //Set<String> matchingUrlPatterns = requestMapping.getPatternsCondition().getPatterns();
```
Comment (kekey1) [inline chpl/chpl-api/src/main/java/gov/healthit/chpl/filter/DeprecatedEndpointUsageFilter.java]: Remove?

File: chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/annotation/CacheControlHandlerInterceptor.java [hunk 3]
```diff
@@ -141,13 +140,15 @@ protected final CacheControl getCacheControl(
...
     }
+
     @Override
     public final boolean preHandle(
             final HttpServletRequest request,
             final HttpServletResponse response,
             final Object handler) throws Exception {
         this.assignCacheControlHeader(request, response, handler);
-        return super.preHandle(request, response, handler);
+        //return super.preHandle(request, response, handler);
```
Comment (kekey1) [inline chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/annotation/CacheControlHandlerInterceptor.java]: Remove?

File: chpl/chpl-service/pom.xml [hunk 4]
```diff
@@ -378,11 +403,18 @@
             <artifactId>spring-boot</artifactId>
             <version>${org.springboot.version}</version>
         </dependency>
+        <!--
```
Comment (kekey1) [inline chpl/chpl-service/pom.xml]: Can this be deleted?

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/entity/datatypes/JsonPostgreSQLDialect.java [hunk 5]
```diff
@@ -1,12 +1,8 @@
...
-
-import org.hibernate.dialect.PostgreSQLDialect;
-
-public class JsonPostgreSQLDialect extends PostgreSQLDialect {
-    public JsonPostgreSQLDialect() {
-        super();
-        this.registerColumnType(Types.JAVA_OBJECT, "json");
-    }
-}
+//public class JsonPostgreSQLDialect extends PostgreSQLDialect {
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/entity/datatypes/JsonPostgreSQLDialect.java]: Can we delete this file?

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/onetime/FixupDeveloperAttestationActivityJob.java [hunk 6]
```diff
@@ -2,8 +2,8 @@
 
 import java.util.List;
 
-import javax.persistence.Query;
-import javax.transaction.Transactional;
+import jakarta.persistence.Query;
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/onetime/FixupDeveloperAttestationActivityJob.java:5]: Could we delete this job instead? And any other onetime jobs?

**Review by kekey1 — DISMISSED** [review]

**Review by kekey1 — DISMISSED** [review]

**Review by andlar — DISMISSED** [review]

**Review by kekey1 — APPROVED** [review]

**Review by andlar — DISMISSED** [review]

**Review by kekey1 — COMMENTED** [review]
I think this needs a change in SearchDevelopersController. Can you check out the feature branch and make sure it compiles?

**Review by kekey1 — CHANGES_REQUESTED** [review]
I think this needs a change in SearchDevelopersController. Can you check out the feature branch and make sure it compiles?

**Review by andlar — APPROVED** [review]

**Review by kekey1 — APPROVED** [review]

**Pull request merged**